Serial RS-232 and RS-485 (#15)
hostmobility/documentation: docs/interfaces/serial.md

---

> title: Serial Communication

Reviewer: ```suggestion
title: Serial communication
```

tags:
  - RS-232
  - RS-485
  - MX-V
  - CT
  - CT GL
  - C61
---
## Overview

> Many of the products has one or more serial communications:

Reviewer: ```suggestion
Several products have one or more serial interfaces:
```

- RS-232
- RS-485

## Platform specific
- [MX-V](mxv/serial.md)
- [CT](mx4/serial.md)
- [CT GL](mx4/serial.md)
- [C61](c61/serial.md)

## Example
```bash
    echo -e hello\r\n | microcom  -s 9600 -t 10000 -X /dev/ttyTHS1
```

## Console output

> It is possible to switch console output (default is one uart that is connected to a debug pin list inside the housing ) to the external Rs-232 from u-boot environment list. Or at linux run time with `setconsole`

Reviewer: ```suggestion
The console output on the device defaults to a UART connected to a debug pin list inside the housing. However, it is possible to switch the console output to an external RS-232 either from the u-boot environment list or dynamically at runtime using `setconsole`.
```

## Baudrate

> All support from 9600 to 115200 baudrate.

Reviewer: ```suggestion
The supported range is from 9600 to 115200.
```
